### ISSUE TITLE
ios-webview is shown as just "ios" 

### ISSUE THREAD

**Issue by RusseII** [DamonOehlman/detect-browser#110]
![image](https://user-images.githubusercontent.com/15036618/58040706-12fa7400-7b04-11e9-8245-fe648c4ed799.png)

I believe this should be detected as ios-webview

**Comment by DamonOehlman** [comment]
Thanks @RusseII - would you mind dropping the UA that the webview is reporting into this issue and I can do some further testing to get to the bottom of this please?

I think some folks have used sites like https://www.whatismybrowser.com/detect/what-is-my-user-agent to achieve this in the past.  I think you should be able to select and copy and paste the text into this issue.

Thanks heaps,
Damon.

**Comment by RusseII** [comment]
Perfect! Thanks @Damon

Mozilla/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit/605.1.15
(KHTML, like Gecko) Version/12.1 Mobile/15E148 Safari/604.1

That is the user agent I got from the site you sent.

**Comment by RusseII** [comment]
Here is the UA I got from the site while in real safari ... it looks like
they are exactly the same?

Real Safari
Mozilla/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit/605.1.15
(KHTML, like Gecko) Version/12.1 Mobile/15E148 Safari/604.1

Webview Safari inside gmail app:
Mozilla/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit/605.1.15
(KHTML, like Gecko) Version/12.1 Mobile/15E148 Safari/604.1

**Comment by RusseII** [comment]
@DamonOehlman anymore thoughts on this?

**Comment by DamonOehlman** [comment]
@RusseII apologies got buried with other things.  They do indeed look identical.  My familiarity with ios has dropped off in recent years.  Perhaps they have lifted their game on apps having the real mobile browser rather than a webview that is lagging behind (which probably causes other problems for app developers).

**Comment by Fjohn666** [comment]
Hello, We have the same problem here with Safari. The browser is detected as "ios". Maybe the regex need to be updated ?

**Comment by RusseII** [comment]
@Fjohn666 It seems that the current method of detection will not work as the User Agent for native Safari, and a Safari webview seem to be identical.

**Comment by DamonOehlman** [comment]
Yep (and thanks for commenting @RusseIl) I think we have hit some limitations with iOS detection.  If anyone comes up with any ideas then we can definitely consider implementing them.

**Comment by RusseII** [comment]
I am not sure how to do it with purely a regex, but right now I'm using this library https://github.com/muaz-khan/DetectRTC, which lets me identify if someone is using a safari webview with: 
```
switch (DetectRTC && DetectRTC.osName) {
    case 'iOS':
      if (browser.name === 'Safari') {
        if (!DetectRTC.isGetUserMediaSupported){
          console.log("Safari Webview Detected!")
}}
```

Then you can see how they check isGetUserMediaSupported here: https://github.com/muaz-khan/DetectRTC/blob/master/dev/DetectRTC.js

Maybe we can do something similar to this approach?

**Comment by DamonOehlman** [comment]
@RusseII nice workaround :)

I think rather than try to implement any kind of detection hack into `detect-browser` itself, encouraging folks to use a technique similar to what you are doing above is best.  I should probably add something to the README about this - probably a link to this issue and specifically your comment above is ideal.
